### PR TITLE
Implement deno.mkdir()

### PR DESCRIPTION
**Pull request by sijad**
related to #593

### CODE REVIEW

File: src/handlers.rs [hunk 1]
```diff
@@ -411,21 +411,30 @@ fn handle_make_temp_dir(d: *const DenoC, base: &msg::Base) -> Box<Op> {
...
   if !deno.flags.allow_write {
     return odd_future(permission_denied());
   }
 
-  // TODO(ry) use blocking
   Box::new(futures::future::result(|| -> OpResult {
     // TODO(ry) Use mode.
     deno_fs::mkdir(Path::new(path))?;
-    Ok(None)
+    let builder = &mut FlatBufferBuilder::new();
```
Comment (sijad) [inline src/handlers.rs]: this probably is not right, please let me know how should it be refactored.

Comment (ry) [reply src/handlers.rs]: `Ok(None)` doesn't work?

File: js/mkdir_test.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import * as deno from "deno";
+
+testPerm({ write: true }, function mkdirSyncSuccess() {
+  const path = deno.makeTempDirSync() + "/dir/subdir";
+  deno.mkdirSync(path);
+  const pathInfo = deno.statSync(path);
+  assert(pathInfo.isDirectory());
+});
+
+testPerm({ write: false }, function mkDirSyncPerm() {
```
Comment (ry) [inline js/mkdir_test.ts]: capitalization should be mkdirSyncPerm

File: js/mkdir.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as fbs from "gen/msg_generated";
+import { flatbuffers } from "flatbuffers";
+import * as dispatch from "./dispatch";
+
+/**
+ * Creates a new directory with the specified path and permission synchronously.
+ *
+ *     import { mkdirSync } from "deno";
+ *     mkdirSync('/new_dir', 0o700);
```
Comment (ry) [inline js/mkdir.ts]: Leave out the permission parameter in the examples.

File: src/handlers.rs (hunk 1 above)
Comment (sijad) [reply src/handlers.rs]: that's what I'm not sure about.
I've tried `Ok(None)` first but this error thrown:

```
Expecting promise in table. 0
```

https://github.com/denoland/deno/blob/35bc9ddf636734a424b8f69ca7a49af071193002/js/dispatch.ts#L16

Comment (sijad) [reply src/handlers.rs]: seems `Ok(None)` does not return back `cmd_id` to js, I thought it's related to `// TODO(ry) use blocking`, so i've come up with this solution.

**Comment by ry** [comment]
@sijad I've added a commit that fixes the bug.

**Pull request merged**